### PR TITLE
Admin Page: Allow subscribers to connect to WP.com

### PR DESCRIPTION
**Pull request by ebinnion**
Subscribers were allowed access to the admin page to connect until #2441.

Now that we've modified SSO to rely on a connection to WP.com, we should allow subscribers access to the admin page so that they can connect their remote account to WordPress.com.

To test:
- Checkout `update/subscriber-allowed-admin-page` branch
- Log in on the remote site with a subscriber account
- Go to Jetpack admin page.
- Link account to WP.com
- Go to dasboard and make sure unlink widget is not there
- Go to Jetpack admin page and attempt to unlink
- Does everything work? Good 😄 

cc @lezama. Also, @dereksmart who originally removed the functionality. Also, for his help in figuring out where to update this header text for subscriber users:

d42dc974-1e82-11e6-982f-b0159522bd24

-------------------
- [ ] Make sure your changes respect [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- [ ] Did you make changes, or create a **new .js file**? If [Gulp](http://gulpjs.com/) is installed on your testing environment, run `gulp js:hint` before to commit your changes. It will allow you to [detect errors in Javascript files](http://jshint.com/about/).
- [ ] Did you create a **new action or filter**? [Add inline documentation](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#4-hooks-actions-and-filters) to help others understand how to use the action or the filter.
- [ ] Create **[unit tests](https://github.com/Automattic/jetpack/tree/master/tests)** if you can. If you're not familiar with Unit Testing, you can check [this tutorial](https://pippinsplugins.com/series/unit-tests-wordpress-plugins/).

### CODE REVIEW

**Comment by dereksmart** [comment]
Fine with me.  I think SSO is the only thing that a subscriber would ever be able to use in Jetpack, yeah?

**Comment by ebinnion** [comment]
I just pushed a commit which customizes the header text for subscribers:

subscriber_non_admin

Same page for other non-admins:

non_admin

**Comment by dereksmart** [comment]
LGTM! we will in the future conditionally show them what they can't do based on whether or not the modules are active, but this is 👌  for now

File: _inc/client/components/non-admin-view/index.jsx [hunk 1]
```diff
@@ -18,7 +18,9 @@ const NonAdminView = React.createClass( {
 	renderContent: function() {
 		const userData = window.Initial_State.userData.currentUser;
 		const isLinked = this.props.isLinked( this.props );
-		let headerText = 'Write posts via email, get notifications about your site activity, and log in with a single click.';
+		let headerText = userData.permissions.edit_posts
+			? 'Write posts via email, get notifications about your site activity, and log in with a single click.'
```
Comment (lezama) [inline _inc/client/components/non-admin-view/index.jsx]: don't we have i18n ?

File: _inc/client/components/non-admin-view/index.jsx [hunk 2]
```diff
@@ -18,7 +18,9 @@ const NonAdminView = React.createClass( {
 	renderContent: function() {
 		const userData = window.Initial_State.userData.currentUser;
 		const isLinked = this.props.isLinked( this.props );
-		let headerText = 'Write posts via email, get notifications about your site activity, and log in with a single click.';
+		let headerText = userData.permissions.edit_posts
```
Comment (lezama) [inline _inc/client/components/non-admin-view/index.jsx]: looks like `headerText` could be a `const`

**Comment by lezama** [comment]
:shipit:

File: _inc/client/components/non-admin-view/index.jsx (hunk 1 above)
Comment (ebinnion) [inline _inc/client/components/non-admin-view/index.jsx]: We chatted in Slack and Derek gave us some helpful links. I'd like to address in a separate PR though.

**Pull request merged**